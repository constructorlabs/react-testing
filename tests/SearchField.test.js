import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import SearchField from '../src/SearchField';

describe('SearchField', () => {
  it('correctly renders tree', () => {
    const tree = renderer.create(<SearchField />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('captures and renders user input', () => {
    const event = {
      target: {
        value: 'Barbara Streisand'
      }
    };
    const wrapper = shallow(<SearchField />);
    wrapper.find('input').simulate('change', event);
    const value = wrapper.find('input').props().value;
    expect(value).toBe('Barbara Streisand');
  });

  it('calls receiver prop function with value state', () => {
    const receiver = jest.fn();
    const event = {
      preventDefault: jest.fn()
    };

    const wrapper = shallow(<SearchField receiver={receiver} />);
    wrapper.setState({ value: 'Paul Anka' });
    wrapper.find('form').simulate('submit', event);

    expect(receiver.mock.calls).toEqual([
      ['Paul Anka']
    ]);

    expect(event.preventDefault.mock.calls).toEqual([ [] ]);
  });
});
