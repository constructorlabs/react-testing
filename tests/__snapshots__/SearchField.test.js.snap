// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchField correctly renders tree 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    type="submit"
  >
    Submit
  </button>
</form>
`;
